### ISSUE TITLE
[DECOHack] Actor Self-Copying Removes State Information

### ISSUE THREAD

**Issue by MTrop** [MTrop/DoomTools#52]
This block of DECOHack code from **Boris** at DoomWorld copies the Imp slot to itself and modifies some fields:
```
#include <doom19>
#include <friendly>


thing MTF_IMP : thing MTF_IMP "S﻿trong Imp"
{
    health 120
    speed 12
}
```

However, it results in the following patch:

```
Patch File for DeHackEd v3.0
# Created with DECOHack v0.17.0 by Matt Tropiano
# Note: Use the pound sign ('#') to start comment lines.

Doom version = 19
Patch format = 6


Thing 12 (Strong Imp)
Hit points = 120
Speed = 12
Initial frame = 0
First moving frame = 0
Injury frame = 0
Close attack frame = 0
Far attack frame = 0
Death frame = 0
Exploding frame = 0
Respawn frame = 0
```

...which does not seem right to me (and to him), as the states were never cleared nor redefined. Maybe something in the parser is clearing those states after copy? Was this introduced during the state parse revamp? Just ignoring the self-copy is probably not the best solution - some code is getting tripped that shouldn't be!

**Comment by MTrop** [comment]
A followup: I can confirm this doesn't happen for copies from one slot to another. Weird!

**Comment by MTrop** [comment]
Not so weird: The following code in `DEHThing.java:copyFrom(DEHThing)`, lines 128 to 130:

```Java
clearLabels();
for (String label : source.getLabels())
    setLabel(label, source.getLabel(label));
```

It makes sense that the destination should clear its label mapping to receive the source's labels in whole, however, if the source is the same Thing as the destination, it clears the source's labels before the copy! Maybe this part should be skipped if they're the same Thing? Would that harm anything? No, right?

**Comment by MTrop** [comment]
Fixed in commit 427a28c8f9b715087ffcc451587cdbb2c7c3b347. Copying the same object now results in a no-op across all objects.
